app/scripts/modules/kubernetes/pipeline/stages/runJob/runJobStage.js

```diff
@@ -52,4 +76,22 @@ module.exports = angular.module('spinnaker.core.pipeline.stage.kubernetes.runJob
     this.onChange = (changes) => {
       this.stage.container.imageDescription.registry = changes.registry;
     };
+
+    this.triggerImages = () => {
+      if (this.pipeline.triggers.length <= 0) return {};
+
+      return this.pipeline.triggers.filter((trigger) => {
+        return trigger.type === 'docker' && trigger.enabled === true;
+      }).map((trigger) => {
+        return `${trigger.registry}/${trigger.repository}`;
+      });
+    };
+
+    this.updateContainerImage = () => {
```

Comment:
This function also needs to update `this.stage.container.imageDescription.account`. I have two Docker Registry accounts, and I can get in a state where the account does not match the container.